storage/plugins/digital-library/DigitalLibraryPlugin.php

```
@@ -195,37 +238,101 @@ private function registerHooks(): void
+            throw $e;
+        }
+    }
+
+    /**
+     * Detect both autocommit(false) and an explicit begin_transaction().
+     * The latter leaves @@autocommit at 1, so a disposable savepoint probe is
+     * required to avoid silently committing the caller's transaction.
+     */
+    private function hasActiveTransaction(): bool
+    {
+        $db = $this->db;
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)